estate/models/estate_property.py

```diff
@@ -150,3 +150,9 @@ def action_cancel(self):
             "Selling price must be strictly positive",
         )
     ]
+
+    @api.ondelete(at_uninstall=False)
+    def unlink_if_not_new_or_cancelled(self):
```

Comment:
If we look at the name of the method and then at the if statement, it's say two different things

The title says that we need to "unlink the record if it is not new or cancelled"

Then the line 157 says that, if the state of the record is not in new or cancelled, we cannot unlink the record

We need to change the declaration of the method, [here are two ways](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.api.ondelete):

> the method should be named either _unlink_if_<condition> or _unlink_except_<not_condition>

one:

```suggestion
    def unlink_if_new_or_cancelled(self):
```

two:

```suggestion
    def  _unlink_except_not_new_or_cancelled(self):
```